File: LeetCode_Marathons/LCM4/Medium/Beautiful_Arrangement_II/a.java
class Solution {
    public int[] constructArray(int n, int k) {
        int[] ans = new int[n];
        int index = 0;
        int l = 1, r = k + 1;
        while (l <= r) {
            ans[index++] = l++;
            if (l > r) break;
            ans[index++] = r--;
        }
        for (int i = k + 2; i <= n; i++) {
            ans[index++] = i;
        }
        return ans;
    }
}

File: LeetCode_Marathons/LCM4/Medium/Beautiful_Arrangement_II/a.py
class Solution:
    def constructArray(self, n: int, k: int) -> List[int]:
        ans: List[int] = []
        l, r = 1, k + 1
        while l <= r:
            ans = ans + [l]
            l += 1
            if l > r: break
            ans = ans + [r]
            r -= 1
        for i in range(k + 2, n + 1): ans = ans + [i]
        return ans